src/univariate/continuous/triangular.jl

```diff
@@ -93,20 +93,28 @@ entropy(d::TriangularDist{T}) where {T<:Real} = one(T)/2 + log((d.b - d.a) / 2)
 
 function pdf(d::TriangularDist, x::Real)
     a, b, c = params(d)
-    return if x < c
-        2 * max(x - a, 0) / ((b - a) * (c - a))
-    else
-        2 * max(b - x, 0) / ((b - a) * (b - c))
+    if x < a || b < x
+        return 0
+    elseif a <= x < c
+        return 2 * (x - a) / ((b - a) * (c - a))
+    elseif x == c
+        return 2 / (b - a)
+    elseif c < x <= b
+        return 2 * (b - x) / ((b - a) * (b - c))
```

Comment:
I'm pretty sure that was (roughly) the original code before someone (I?) tried to make it type stable and apparently introduced bugs which were not covered by the tests (even though we compare it with R, IIRC). Unfortunately, this PR has the same type stability issues as the original versions which causes the test errors.

Reply:
It seems the current version is correct apart from degenerate cases where `a == c` or `b  == c` since in these cases we would divide by zero.

Reply:
Ok, thanks. Can you point me to where the test for type stability is performed? I can try to modify for type stability, would be helpful to be able to test it directly.

Reply:
It's part of the large test suite for univariate distributions, there's no simple TriangularDist-specific test. For instance, type stability checks are performed in https://github.com/JuliaStats/Distributions.jl/blob/c431d20d3a0f00e0cf22f9fd959b9bb11d978517/test/testutils.jl#L487, which is part of https://github.com/JuliaStats/Distributions.jl/blob/2dc764e450a5b33f1506669e3649f37a1dbda4f1/test/testutils.jl#L69, which is called by https://github.com/JuliaStats/Distributions.jl/blob/c431d20d3a0f00e0cf22f9fd959b9bb11d978517/test/univariates.jl#L51.

But locally something like
```julia
using Test
@inferred(pdf(TriangularDist(0f0, 2f0, 1f0), 1.0))
```
or just
```julia
@code_warntype pdf(TriangularDist(0f0, 2f0, 1f0), 1.0)
```
should be sufficient to reproduce the errors with the `pdf` code in this PR: In one branch the function would return `Int`, in one branch `Float32`, and in the other branches `Float64`.

Reply:
Thanks for the help. Should be a very educational exercise for me 🙂

Reply:
I have updated to be type stable now. All the tests passed on my local. I think this should be good to go now but let me know if anything else is required.

Reply:
Unfortunately, it's still not type stable.